gilded-rose-kata/src/main/kotlin/models/Item.kt

```diff
@@ -0,0 +1,25 @@
+package models
+
+import enums.*
+import interfaces.*
+
+abstract class Item(val name: String, var sellIn: Int, quality: Int): ItemQualityModifier {
```

Comment:
This class hierarchy looks interesting. Is an `Item` also a `ItemQualityModifier`? Can a `ItemQualityModifier` modify other items or only itself? If only itself, shouldn't this be an implementation detail of the class?

Reply:
First of all: Speaking what was in my mind!
My goal with this interface was to make clear that every `Item` has an intrinsic behaviour showing that it would be `Modified`. The naming could be better (now that i'm reading it again) and could be more closely related to timing since it has `onDayTick() and onFinalDay()` methods.

Is any part of this mind modeling bad?